### ISSUE TITLE
Caffe got stuck halfway when training imagenet dataset

### ISSUE THREAD

**Issue by liangshuang1993** [BVLC/caffe#5954]
Please use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) for usage, installation, or modeling questions, or other requests for help.
_Do not post such requests to Issues._ Doing so interferes with the development of Caffe.

Please read the [guidelines for contributing](https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md) before submitting this issue.

### Issue summary
I'm training imagenet dataset using inception v3. Everything was fine in the first two days, but then caffe was stuck since yesterday.  The log hasn't been updated for 24 hours, but when I check cpu and gpu, it seems that caffe in still runing. Does anyone know what's going on? Thanks a lot!

### nvidia-smi output:
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla K80           On   | 0000:08:00.0     Off |                    0 |
| N/A   51C    P0   124W / 149W |  10200MiB / 11439MiB |     88%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla K80           On   | 0000:09:00.0     Off |                    0 |
| N/A   67C    P0   134W / 149W |   6412MiB / 11439MiB |     80%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla K80           On   | 0000:83:00.0     Off |                    0 |
| N/A   53C    P0   136W / 149W |   6412MiB / 11439MiB |     92%      Default |

### log
...
0: I0928 08:55:50.993021 58717 sgd_solver.cpp:105] Iteration 113320, lr = 0.0088668
0: I0928 08:56:26.529989 58717 solver.cpp:218] Iteration 113360 (1.12558 iter/s, 35.5371s/40 iters), loss = 0.529116
0: I0928 08:56:26.530251 58717 solver.cpp:237]     Train net output #0: loss1/loss1 = 1.59655 (* 0.3 = 0.478965 loss)
0: I0928 08:56:26.530263 58717 sgd_solver.cpp:105] Iteration 113360, lr = 0.0088664
0: I0928 08:57:02.026926 58717 solver.cpp:218] Iteration 113400 (1.12686 iter/s, 35.4968s/40 iters), loss = 0.559563
0: I0928 08:57:02.027178 58717 solver.cpp:237]     Train net output #0: loss1/loss1 = 1.18278 (* 0.3 = 0.354835 loss)
0: I0928 08:57:02.027192 58717 sgd_solver.cpp

**Comment by ngmq** [comment]
I am experiencing the same problem with the latest code on master branch cloned with git yesterday.

In my case, the training reaches around 23000 iterations and then hangs. Once it hangs there is no logs anymore. I am using a combination of lmdb layer and MemoryDataLayer.

The configuration of 02 data layers of my model, if that helps:

name: "MyNet"
layer {
    name: "lowlevel_db"
    type: "Data"
    top: "lowdata"
    top: "label"
    transform_param {
	    mirror: false
    } 
    data_param {
        source: "train_lowDB"
        backend: LMDB
        batch_size: 100
    }
}
layer {
  name: "highlevel_db"
  type: "MemoryData"
  top: "conv5_3"
  top: "fakelabel"
  transform_param {
	mirror: false
  }
  memory_data_param {
	batch_size: 100
	channels: 512
	height: 14
	width: 14
  }
}
